File: frontend/src/types/flow.ts
export interface Flow {
    id: string
    project_id: string
    title: string
    description: string
    created_at: string
    updated_at: string
  }
  
  export interface CreateFlowBody {
    project_id: string
    title: string
    description: string
  }
  
  export interface UpdateFlowBody {
    title?: string
    description?: string
  }
  
  export interface GetFlowsResponse {
    total_count: number
    page: number
    flows: Flow[]
  }
  
  

File: fastapi-ca/flow/infra/db_models/flow.py
from datetime import datetime
from sqlalchemy import String, DateTime, Text
from sqlalchemy.orm import Mapped, mapped_column
from database import Base

class Flow(Base):
    __tablename__ = "Flow"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    project_id: Mapped[str] = mapped_column(String(36), nullable=False)
    title: Mapped[str] = mapped_column(String(64), nullable=False)
    description: Mapped[str] = mapped_column(String(64), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
